include/osmium/io/writer.hpp

```diff
@@ -236,9 +243,12 @@ namespace osmium {
                     (set_option(options, args), 0)...
                 };
 
-                if (options.header.get("generator") == "") {
-                    options.header.set("generator", "libosmium/" LIBOSMIUM_VERSION_STRING);
-                }
+                if (options.pool)
+                    m_pool = options.pool;
+                else
+                    m_pool = &thread::Pool::default_instance();
+
+                m_output = osmium::io::detail::OutputFormatFactory::instance().create_output(*m_pool, m_file, m_output_queue);
```

Comment:
Your concern seems similar to the situation discussed here:
https://akrzemi1.wordpress.com/2012/11/14/not-using-stdthread/#comment-659

Isn't `Writer::m_thread` considered to actually be *"launching of the thread is the last instruction in the constructor body"* (if we shuffle `ensure_cleanup` a bit) ?